uaclient/tests/test_apt.py

```diff
@@ -152,7 +152,34 @@ def test_errors_on_process_execution_errors(
         apt_helper_call = mock.call(
             ['/usr/lib/apt/apt-helper', 'download-file',
              'http://user:pwd@fakerepo/ubuntu/pool/',
-             '/tmp/uaclient-apt-test'])
+             '/tmp/uaclient-apt-test'], timeout=20)
+        assert [apt_helper_call] == m_subp.call_args_list
+        assert [mock.call('/tmp/uaclient-apt-test')] == m_unlink.call_args_list
+
+    @mock.patch('uaclient.apt.os.unlink', return_value=True)
+    @mock.patch('uaclient.util.subp')
+    @mock.patch('uaclient.apt.os.path.exists', return_value=True)
+    def test_errors_on_apt_helper_process_timeout(
+            self, m_exists, m_subp, m_unlink):
+        """Raise the appropriate user facing error from apt-helper timeout."""
+
+        # Failure apt-helper response
+        m_subp.side_effect = exceptions.CommandTimeoutError(
+            cmd=['apt-helper', 'SECRETS'], timeout=10)
+        with pytest.raises(exceptions.CommandTimeoutError) as excinfo:
+            assert_valid_apt_credentials(
+                repo_url='http://fakerepo', username='user', password='pwd')
+        error_msg = (
+            'Timeout after 20 seconds: cannot validate credentials for APT'
```

Comment:
This error message feels the wrong way around to me; it's structured as "specific: general". I think something more like "Cannot validate credentials for APT: timeout after 20 seconds" feels clearer to me, as it's telling me what failed and then giving me the specific reason for that failure.